server/container_create_linux.go

```diff
@@ -282,9 +291,13 @@ func (s *Server) createSandboxContainer(ctx context.Context, ctr ctrIface.Contai
 
 	skipRelabel := false
 	const superPrivilegedType = "spc_t"
-	if securityContext.SelinuxOptions.Type == superPrivilegedType || // super privileged container
-		(ctr.SandboxConfig().Linux.SecurityContext.SelinuxOptions.Type == superPrivilegedType && // super privileged pod
-			securityContext.SelinuxOptions.Type == "") {
+	if securityContext.SelinuxOptions != nil &&
+		(securityContext.SelinuxOptions.Type == superPrivilegedType || // super privileged container
+			(ctr.SandboxConfig().Linux != nil &&
+				ctr.SandboxConfig().Linux.SecurityContext != nil &&
+				ctr.SandboxConfig().Linux.SecurityContext.SelinuxOptions != nil &&
+				ctr.SandboxConfig().Linux.SecurityContext.SelinuxOptions.Type == superPrivilegedType && // super privileged pod
+				securityContext.SelinuxOptions.Type == "")) {
```

Comment:
the tests seem to be failing because skipRelabel should be true:
https://ci.openshift.redhat.com/jenkins/job/test_pull_request_crio_integration_crun_fedora/17060/s3/


I gave it a shot myself, and I found we'll need something like
```diff
diff --git a/server/container_create_linux.go b/server/container_create_linux.go
index 561e43eb7d..17caa935a3 100644
--- a/server/container_create_linux.go
+++ b/server/container_create_linux.go
@@ -291,13 +291,15 @@ func (s *Server) createSandboxContainer(ctx context.Context, ctr ctrIface.Contai
 
 	skipRelabel := false
 	const superPrivilegedType = "spc_t"
-	if securityContext.SelinuxOptions != nil &&
-		(securityContext.SelinuxOptions.Type == superPrivilegedType || // super privileged container
-			(ctr.SandboxConfig().Linux != nil &&
-				ctr.SandboxConfig().Linux.SecurityContext != nil &&
-				ctr.SandboxConfig().Linux.SecurityContext.SelinuxOptions != nil &&
-				ctr.SandboxConfig().Linux.SecurityContext.SelinuxOptions.Type == superPrivilegedType && // super privileged pod
-				securityContext.SelinuxOptions.Type == "")) {
+	if securityContext.SelinuxOptions == nil {
+		securityContext.SelinuxOptions = &types.SELinuxOption{}
+	}
+	if securityContext.SelinuxOptions.Type == superPrivilegedType || // super privileged container
+		(ctr.SandboxConfig().Linux != nil &&
+			ctr.SandboxConfig().Linux.SecurityContext != nil &&
+			ctr.SandboxConfig().Linux.SecurityContext.SelinuxOptions != nil &&
+			ctr.SandboxConfig().Linux.SecurityContext.SelinuxOptions.Type == superPrivilegedType && // super privileged pod
+			securityContext.SelinuxOptions.Type == "") {
 		skipRelabel = true
 	}
 
```

Reply:
Ah, great catch! I applied the patch as suggested :+1: